mungegithub/mungers/submit-queue.go

```diff
@@ -1269,6 +1271,46 @@ func (sq *SubmitQueue) servePriorityInfo(res http.ResponseWriter, req *http.Requ
 </ol> `))
 }
 
+func (sq *SubmitQueue) getHealthSVG() []byte {
+	sq.Lock()
+	defer sq.Unlock()
+	blocked := false
+	blockingJobs := make([]string, 0)
+	blocked = !sq.health.MergePossibleNow
+	if blocked {
+		for job, status := range sq.e2e.GetBuildStatus() {
+			if status.Status == "Not Stable" {
+				job = strings.Replace(job, "kubernetes-", "", -1)
+				blockingJobs = append(blockingJobs, job)
+			}
+		}
+	}
+	status := "running"
+	color := "brightgreen"
+	if blocked {
+		status = "blocked"
+		color = "red"
+		sort.Strings(blockingJobs)
+		n := len(blockingJobs)
+		if n > 0 {
+			if n > 3 {
+				n = 3
+			}
+			status += " by " + strings.Join(blockingJobs[:n], ", ")
+			if len(blockingJobs) > 3 {
```

Comment:
combine with above n > 3 if statement, and actually just put "..." in the array so that the strings.Join will format it correctly.